### ISSUE TITLE
Write parqutet  sometimes doesn't clear tasks

### ISSUE THREAD

**Issue by gabrielelanaro** [dask/dask#3402]
The following code shows that fastparquet tasks are kept in memory (instead of being executed in a streaming fashion).

I was wondering if this is the expected behaviour

dask (0.17.2)
distributed (1.21.6)

```
import pandas as pd
import dask.dataframe as dd
import numpy as np
from dask.distributed import Client
import time

val= np.random.rand(3000)
df = dd.from_pandas(pd.DataFrame(val, columns=['a']), npartitions=1000)

def wait(df):
    time.sleep(1)
    return df

df = df.map_partitions(wait, meta=df)

c = Client()

df.to_parquet('/tmp/test', engine='pyarrow')
```

This is how the task execution looks like:

![screenshot from 2018-04-13 09-24-17](https://user-images.githubusercontent.com/239875/38747349-7b088fb0-3eff-11e8-808d-d53dc6114dd0.png)
![screenshot from 2018-04-13 09-41-37](https://user-images.githubusercontent.com/239875/38747353-7df6ae46-3eff-11e8-98c2-0171acffac2e.png)

related issue: #2470

**Comment by martindurant** [comment]
Yes, it is expected that the tasks are kept in memory. The metadata from each is being gathered, so that a final summary `_metadata` file can be written.

**Comment by mrocklin** [comment]
It's worth noting that this metadata is very small in memory

**Comment by martindurant** [comment]
> It's worth noting that this metadata is very small in memory

...except in the pathological case of very many columns and very few rows per partition.
